src/main/java/uk/gov/ons/ctp/integration/contactcentresvc/service/impl/CaseServiceImpl.java

```diff
@@ -1059,4 +1057,54 @@ private void sendEvent(EventType eventType, EventPayload payload, Object caseId)
         .with("transactionId", transactionId)
         .debug("{} event published", eventType);
   }
+
+  private CaseDTO getLatestCaseById(UUID caseId, Boolean getCaseEvents) throws CTPException {
+
+    List<CaseDTO> cases = new ArrayList<>();
+    CaseContainerDTO caseFromRM = null;
+    CaseDTO caseDto = null;
+
+    try {
+      caseFromRM = getCaseFromRm(caseId, getCaseEvents);
+    } catch (ResponseStatusException ex) {
+      if (ex.getStatus() == HttpStatus.NOT_FOUND) {
+        log.with("caseId", caseId).debug("Case Id Not Found by Case Service");
+      } else {
+        log.with("caseId", caseId)
+            .with("status", ex.getStatus())
+            .error("Error calling Case Service");
+        throw ex;
+      }
+    }
+
+    if (caseFromRM != null) {
+      caseDto = mapCaseContainerDTO(caseFromRM);
```

Comment:
Try to declare variables as close as possible to where they are used. 
In this instance:
`CaseDTO caseDto = mapCaseContainerDTO(caseFromRM);`
and remove the declaration further up.

Apply the "closest to where used" to all the code you write. The other declarations could be closer too.

Reply:
good spot - I was originally using the caseDto variable for 2 different purposes, which is why it needed to be declared outside the if block, but I am now only using it here so it makes sense to declare it here as you suggest. Done.